src/Nethermind/Nethermind.Core/Transaction.cs

```diff
@@ -257,6 +257,8 @@ public string ToString(string indent)
 
         public bool MayHaveNetworkForm => Type is TxType.Blob;
 
+        public bool IsSystemCall { get; set; } = false;
```

Comment:
use `TransactionExtensions.IsSystem` extension method

Reply:
The problem is that IsSystem != IsSystemCall. Gas estimation uses system tx by default and expect gas spendings to be the same as for regular tx. By introducing IsSystemCall we also prevent potential problems for chains that already use system transactions and it's dangerous to change gas calculations for them. Additionally, what's is real expected is "system call" but not real transaction.

Reply:
Hm maybe instead of field we can make a systemcallliketransaction type and check for it